packages/palette/src/elements/Modal/ModalBase.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useRef } from "react"
+import { createPortal } from "react-dom"
+import { FocusOn } from "react-focus-on"
+import styled, { css } from "styled-components"
+import { Flex, FlexProps } from "../Flex"
+
+const ModalWrapper = styled(Flex)<{ overlay?: boolean }>`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  /* TODO: Use a token? Or some other alternative to wildstyle z-index? */
+  z-index: 9999;
+
+  ${({ overlay }) =>
+    overlay &&
+    css`
+      /* TODO: Use a token? */
+      background-color: rgba(229, 229, 229, 0.5);
+    `}
+`
+/**
+ * BaseModal
+ */
+export type ModalBaseProps = FlexProps & {
+  overlay?: boolean
+  onClose?(): void
+}
+
+/**
+ * BaseModal
+ * Low-level modal that has no opinions about layout.
+ * Modals content using a portal, locks focus, handles aria-hidden isolation.
+ */
+export const ModalBase: React.FC<ModalBaseProps> = ({
+  children,
+  overlay = true,
+  onClose = () => null,
+  ...rest
+}) => {
+  const appendEl = useRef(document.createElement("div"))
+  const wrapperEl = useRef<HTMLDivElement | null>(null)
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    if (event.target === wrapperEl.current) {
+      onClose()
+    }
+  }
+
+  useEffect(() => {
+    const { current } = appendEl
+    document.body.appendChild(current)
+    return () => {
+      document.body.removeChild(current)
+    }
+  }, [])
+
+  return createPortal(
+    <FocusOn onClickOutside={onClose} onEscapeKey={onClose}>
+      <ModalWrapper
+        // TODO: Remove cast as any. Something is wrong with styled's typings?
+        ref={wrapperEl as any}
```

Comment:
Not sure what's up with this since `styled.whatever` should be able to accept a ref with the underlying DOM type. I'm assuming it's just related to the older version here?

Reply:
Very likely a version conflict. We really need to upgrade styled components but that's a whole thing.

Reply:
FYI, we generally attach portal conteent to [this ID](https://github.com/artsy/force/blob/master/src/desktop/components/main_layout/templates/react_blank_index.jade#L18) (which is spread across a few different templates).

Reply:
@damassi will update — what's the benefit of portaling into a specific DOM node — so it's predictable and ensures there's only ever a single modal? What I don't love about it is that it creates an assumption about the structure of the DOM in this library that we can't guarantee. So the proper thing to do then would be: either it's configurable, or we search for that specific node then fall back to just appending.

Reply:
That sounds good; yeah, this was more just historical info to be aware of, how we handle our current auth models, etc